include/swift/Driver/ToolChain.h

```diff
@@ -87,12 +87,12 @@ class ToolChain {
 
   /// Packs together information chosen by toolchains to create jobs.
   struct InvocationInfo {
-    const char *ExecutableName;
+    StringRef ExecutableName;
```

Comment:
This was deliberately a `char *` to show that it was null-terminated. If you're removing that guarantee, you need to update the uses of it.